### PR TITLE
Adds acceptance tests for product listing

### PR DESCRIPTION
**Pull request by brurez**
This pull request adds acceptance tests to the application. The idea is to cover the application's general use cases. Verify that the application and its parts are interacting correctly and meeting the application requirements.

## How to test
Run `yarn test src/App.test.js`

### CODE REVIEW

**Review by magg01 — APPROVED** [review]
looks good to me. I would like to write some tests for anything further that I add and it's great to have your examples to go off of. Happy to see the back of the .eslintcache file too!

File: src/App.test.js [hunk 1]
```diff
@@ -1,8 +1,126 @@
...
+
+        test("if it renders the new product in product list", () => {
+          const productItem = screen.getAllByText(newProduct.name)[0];
+          expect(productItem).toBeInTheDocument();
+        });
+      });
+    });
+  });
+
+  describe("It is possible to edit an existing product in the list", () => {
```
Comment (dmytrokulak) [inline src/App.test.js:77]: One of the points given to us is for "well documented tests". Huge kudos to including tests, each of use will have to cover our development with tests from now on, but they also have to include comments, I believe.

**Review by bilyanageorgieva — APPROVED** [review]
Very elaborate 👏

File: src/App.test.js (hunk 1 above)
Comment (dmytrokulak) [reply src/App.test.js:98]: I don't know what is their definition of "well documented", but as long as we have comments describing what's going on there - i hope that must be enough. Thank you, Bruno!

**Review by dmytrokulak — APPROVED** [review]

**Comment by dmytrokulak** [comment]
Let's also include [this commit](https://github.com/agiledev3/buyers-remorse-app/commit/608921192e58711f1c5f90fad6d995e755d71651)  into this pull request. It adds test runs into our deployment pipeline.

**Pull request merged**